index.bs

```diff
@@ -6767,17 +6775,56 @@ Note: this extension may be implemented for [=authenticators=] that do not use [
 
         :   <dfn>evalByCredential</dfn>
         ::  A record mapping [=base64url encoding|base64url encoded=] [=credential IDs=] to PRF inputs to evaluate for that credential. Only applicable during [=assertions=] when {{PublicKeyCredentialRequestOptions/allowCredentials}} is not empty.
+
+        :   <dfn>importCryptoKey</dfn>
+        ::  Arguments for the [=client=] to invoke {{SubtleCrypto/importKey}}, along with the PRF result as the `keyData` argument.
+            If present, the extension outputs will be {{Promise}}s resolving to {{CryptoKey}} values;
+            if not present, the extension outputs will be {{BufferSource}} values.
+
+            The client ensures domain separation between {{BufferSource}} and {{CryptoKey}} results,
+            so that an extension input requesting unextractable {{CryptoKey}} values cannot be "downgraded"
+            to requesting the same results as extractable {{BufferSource}} values.
+            For the same reason, the client also ensures domain separation between the [TRUE] and [FALSE] values
```

Comment:
I suspect the domain separation should cover all the parameters? Here's a silly example, but perhaps there are better ones:

A service uses this facility to get an opaque AES-GCM key and claims that, even if the Javascript is compromised, the compromise is limited to existing ciphertexts and ciphertexts that are encrypted in the future, after the compromise is eliminated, are safe again because the key couldn't be extracted.

But if they use a counter for a nonce, which is safe in general, an attacker could tweak the import parameters to get an opaque AES-ECB key and precompute E<sub>k</sub>(x) for the _x_ that will be used by future AES-GCM ciphertexts.

Reply:
Sounds fair. Note that WebCrypto doesn't support AES-ECB, but I suppose you could probably emulate it by feeding single blocks at a time into AES-CBC.

I suppose for that we would need something like a hash of a canonical representation of the arguments - one invariant in order of `EcKeyImportParams` members, for example, or maybe even invariant in `"AES-GCM"` vs. `{name: "AES-GCM"}`.